### ISSUE TITLE
You forgot to wrap your app in <UserProvider>

### ISSUE THREAD

**Issue by Kvek** [auth0/nextjs-auth0#1612]
### Checklist

- [X] The issue can be reproduced in the [nextjs-auth0 sample app](https://github.com/auth0-samples/auth0-nextjs-samples/tree/main/Sample-01) (or N/A).
- [X] I have looked into the [Readme](https://github.com/auth0/nextjs-auth0#readme), [Examples](https://github.com/auth0/nextjs-auth0/blob/main/EXAMPLES.md), and [FAQ](https://github.com/auth0/nextjs-auth0/blob/main/FAQ.md) and have not found a suitable solution or answer.
- [X] I have looked into the [API documentation](https://auth0.github.io/nextjs-auth0/) and have not found a suitable solution or answer.
- [X] I have searched the [issues](https://github.com/auth0/nextjs-auth0/issues) and have not found a suitable solution or answer.
- [X] I have searched the [Auth0 Community](https://community.auth0.com) forums and have not found a suitable solution or answer.
- [x] I agree to the terms within the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md).

### Description

Same as this #1557.

Hi,
I setup a new Next.js project (v14) and have been getting this error after following this guide up to this [step](https://auth0.com/docs/quickstart/webapp/nextjs/interactive#add-the-userprovider-component).

Any help is appreciated. Thanks 😁

### Reproduction

Set up a new app. 
`npx create-next-app auth0-app`
  
Install dependencies
`npm install @auth0/nextjs-auth0`

Add  `.env.local` with the correct appropriate values
```
AUTH0_SECRET='use [openssl rand -hex 32] to generate a 32 bytes value'
AUTH0_BASE_URL='http://localhost:3000'
AUTH0_ISSUER_BASE_URL='https://{yourDomain}'
AUTH0_CLIENT_ID='{yourClientId}'
AUTH0_CLIENT_SECRET='{yourClientSecret}'
```

Add `app/api/auth/[auth0]/route.ts`
```
import { handleAuth } from '@auth0/nextjs-auth0';

export const GET = handleAuth();
```

Add `UseProvider`
```
import { UserProvider } from '@auth0/nextjs-auth0/client';

export default function RootLayout({ children }) {
  return (
    <html lang="en">
      <UserProvider>
        <body>{children}</body>
      </UserProvider>
    </html>
  );
}
```

Run the app
> npm run dev

See this in the browser console.

![Screenshot 2023-12-19 at 09 37 53](https://github.com/auth0/nextjs-auth0/assets/8972543/c80fd866-20e6-4099-94c0-8642ace98ee3)

### Additional context

package.json
```
{
  "name": "auth0-app",
  "version": "0.1.0",
  "private": true,
  "scripts": {
    "dev": "next dev",
    "build": "next build",
    "start": "next start",
    "lint": "next lint"
  },
  "dependencies": {
    "@auth0/nextjs-auth0": "^3.5.0",
    "next": "14.0.4",
    "react": "^18",
    "react-dom": "^18"
  },
  "devDependencies": {
    "@types/node": "^20",
    "@types/react": "^18",
    "@types/react-dom": "^18",
    "eslint": "^8",
    "eslint-config-next": "14.0.4",
    "typescript": "^5"
  }
}

```

layout.tsx
```
import { UserProvider } from '@auth0/nextjs-auth0/client';
import { ReactNode } from 'react';

export default async function RootLayout({ children }:{children:ReactNode}) {
  return (
    <html lang="en">
      <UserProvider>
        <body>{children}</body>
      </UserProvider>
    </html>
  );
}
```

### nextjs-auth0 version

@auth0/nextjs-auth0: "^3.5.0",

### Next.js version

next: "14.0.4",

### Node.js version

v20.10.0

**Comment by Jelenbl** [comment]
having the same issue the `app/api...` and `layout.jsx` files look near identical.

**Comment by dwgardner** [comment]
I was having this same issue and was able to solve it by creating a component named LoginSection.tsx and moving the useUser check into the component.  I referenced the component in the layout.  It seems that you cannot use useUser inside of a layout since there has not been a <UserProvider> tag yet.  By putting it in a component it changes the order of execution.  I won't pretend to understand it completely, but it seems to work.

**Comment by jtw-r** [comment]
> It seems that you cannot use useUser inside of a layout since there has not been a UserProvider tag yet.  By putting it in a component it changes the order of execution.  I won't pretend to understand it completely, but it seems to work.

I believe this is fully intentional, but maybe not documented the best. You cannot use react hooks in server components, (i.e. functions that begin with "use..." or deal with memos, effects, and [context providers or consumers](https://vercel.com/guides/react-context-state-management-nextjs#rendering-third-party-context-providers-in-server-components)). Layout.tsx files are server components--I haven't looked hard, but I don't think Next.js supports them being client components. This is all coming down from the react team if I understand correctly.

Tldr, you want to only call useUser or useAuth in client components.
